### PR TITLE
Upload packages & install scripts to S3

### PR DESCRIPTION
**Pull request by amdprophet**
* Packages & install scripts are now uploaded to AWS S3
* Added new releases workflow
* Added documentation on the release process
* Install scripts use AWS S3 for determining the latest version and fetching packages for macOS & Windows
* The package version is now determined from the otelcol-sumo binary
* Collector artifacts are now fetched exclusively from a remote workflow
* Installs scripts added to a release are now pulled from the correct Git revision

Examples of an upload:

https://sumologic-osc-ci-builds.s3.us-west-2.amazonaws.com/latest_version
https://sumologic-osc-ci-builds.s3.us-west-2.amazonaws.com/0.108.0-1827/install.sh
https://sumologic-osc-ci-builds.s3.us-west-2.amazonaws.com/0.108.0-1827/install.ps1

Signed-off-by: Justin Kolberg <amd.prophet@gmail.com>

### CODE REVIEW

File: packages.cmake [hunk 1]
```diff
@@ -135,6 +68,13 @@ macro(build_cpack_config)
     create_packagecloud_publish_target(${_pc_user} ${_pc_repo} ${_pc_distro} ${_package_output})
   endforeach()
 
+  # Add a target for uploading the package to Amazon S3
+  set(_s3_channel "ci-builds")
+  set(_version "${OTC_VERSION}-${BUILD_NUMBER}")
```
Comment (chan-tim-sumo) [inline packages.cmake:73]: should this be ```OTC_BUILD_NUMBER```? 🤔

Comment (amdprophet) [reply packages.cmake:73]: No, although I can see how that is confusing. The CMake code reads `ENV{OTC_BUILD_NUMBER}` and stores it in a CMake variable called `BUILD_NUMBER`: https://github.com/SumoLogic/sumologic-otel-collector-packaging/blob/upload-s3/version.cmake#L18.

File: .github/workflows/_reusable_build_package.yml [hunk 2]
```diff
@@ -3,12 +3,6 @@ name: '_reusable_build_package'
...
   workflow_call:
     inputs:
-      otc_version:
-        required: true
-        type: string
-      otc_sumo_version:
-        required: true
-        type: string
       otc_build_number:
         required: false
```
Comment (echlebek) [inline .github/workflows/_reusable_build_package.yml:7]: should this be required true? what about workflow_id?

File: version_artifact.cmake [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+set(CMAKE_PLATFORM_INFO_DIR "${CMAKE_SOURCE_DIR}/build/cmake")
+include(CMakeDetermineSystem)
+
+# Set supported platforms for version detection
+set(_supported_platforms "")
+list(APPEND _supported_platforms "darwin_amd64")
+list(APPEND _supported_platforms "darwin_arm64")
+list(APPEND _supported_platforms "linux_amd64")
+list(APPEND _supported_platforms "linux_arm64")
+list(APPEND _supported_platforms "windows_amd64")
```
Comment (echlebek) [inline version_artifact.cmake:39]: Do we support windows_arm64?

File: .github/workflows/_reusable_build_package.yml (hunk 2 above)
Comment (amdprophet) [reply .github/workflows/_reusable_build_package.yml:7]: Yes, good catch.

File: version_artifact.cmake (hunk 3 above)
Comment (amdprophet) [reply version_artifact.cmake:39]: We do not at the moment.

**Pull request merged**